src/main/java/com/microsoft/azure/relay/HybridHttpConnection.java

```diff
@@ -209,24 +209,27 @@ void invokeRequestHandler(RequestCommandAndStream requestAndStream) {
 			return this.listener.sendControlCommandAndStreamAsync(listenerCommand, responseBodyBuffer, timeout)
 					.thenRun(() -> RelayLogger.logEvent("httpSendResponseFinished", this, "control", String.valueOf(responseCommand.getStatusCode())));
 		} else {
+			TimeoutHelper timeRemaining = new TimeoutHelper(timeout);
```

Comment:
>TimeoutHelper timeRemaining = new TimeoutHelper(timeout); [](start = 3, length = 57)

I wouldn't bother changing this now, but keep in mind "timeoutHelper" is really a replacement for "timeout" argument. Any code path which would use timeout should really be aware of timeoutHelper. My personal guideline is therefore to try to keep TimeoutHelper at the top of the method.